src/Data/Generic.purs

```diff
@@ -268,6 +268,24 @@ instance eqGeneric :: Eq GenericSpine where
     eq (SArray xs)  (SArray ys)  = length xs == length ys && zipAll (\x y -> x unit == y unit) xs ys
     eq _ _ = false
 
+eqDataConstructor :: DataConstructor -> DataConstructor -> Boolean
+eqDataConstructor p1 p2 = p1.sigConstructor == p2.sigConstructor
+                       && zipAll (\x y -> x unit == y unit) p1.sigValues p2.sigValues
+
+instance eqGenericSignature :: Eq GenericSignature where
+  eq (SigProd s1 arr1) (SigProd s2 arr2) = s1 == s2
+                                        && length arr1 == length arr2
+                                        && zipAll eqDataConstructor arr1 arr2
+  eq (SigRecord arr1) (SigRecord arr2)   = zipAll (\x y -> x.recLabel == y.recLabel) arr1 arr2
```

Comment:
I think we need to sort by `recLabel` here? The same goes for `Eq GenericSpine` actually, I suppose.